src/Security/Member.php

```diff
@@ -1763,11 +1761,16 @@ public function generateRandomPassword(int $length = 0): string
     {
         $password = '';
         $validator = Member::password_validator();
-        if ($length && $validator && $length < $validator->getMinLength()) {
-            throw new InvalidArgumentException('length argument is less than password validator minLength');
+        if ($validator instanceof RulesPasswordValidator) {
+            $validatorMinLength = $validator->getMinLength();
+            if ($length && $length < $validatorMinLength) {
+                throw new InvalidArgumentException('length argument is less than password validator minLength');
+            }
+        } else {
+            // Make sure the password is long enough to beat even very strict entropy tests
+            $validatorMinLength = 128;
         }
-        $validatorMinLength = $validator ? $validator->getMinLength() : 0;
-        $len = $length ?: max($validatorMinLength, 20);
+        $len = max($length, $validatorMinLength, 20);
```

Comment:
Set a very long password if we're not using the rules-based validator, since that should pass basically any password validation.

Left the rest of the code as-is because using those various characters should make a suitably complex password to pass just about any validation, even custom validation.

Reply:
What does the 128 $validatorMinLength do exactly? That's not saying people need 128 length password right?

Reply:
This is for generating a random password. It ensures that the randomly generated password is at least 128 characters long because that is a very long password and therefore will almost certainly pass and validation logic.